packages/packages.go

```diff
@@ -189,10 +202,90 @@ var ZipFileSystemBuilder = func(p *Package) (PackageFileSystem, error) {
 
 // Init initializes the indexer.
 func (i *FileSystemIndexer) Init(ctx context.Context) (err error) {
-	i.packageList, err = i.getPackagesFromFileSystem(ctx)
+	if err := i.updatePackageFileSystemIndex(ctx); err != nil {
+		i.logger.Error("initializing package filesystem index failed",
+			zap.Error(err),
+			zap.String("indexer", i.label))
+		return err
+	}
+
+	if i.enablePathsWatcher {
+		go i.watchPackageFileSystem(ctx)
+	}
+	return nil
+}
+
+func (i *FileSystemIndexer) watchPackageFileSystem(ctx context.Context) {
```

Comment:
implemented the watcher using fsnotify and working as expected. I would need to validate the debounce time as if is too low the update can't read the manifest of the packages when reading, and if is too high the operation might take too much time where there are lots of packages.

offline discussion about this pr with @mrodm he shared that fsnotify might be not-compatible with certain OS systems. @jsoriano do you know if [this](https://github.com/fsnotify/fsnotify?tab=readme-ov-file#why-dont-notifications-work-with-nfs-smb-fuse-proc-or-sys) could be an issue? an option was to enable a flag/config to set the "mode" of watcher, implement notify but also the active loop? WDYT?

Reply:
> offline discussion about this pr with @mrodm he shared that fsnotify might be not-compatible with certain OS systems. @jsoriano do you know if [this](https://github.com/fsnotify/fsnotify?tab=readme-ov-file#why-dont-notifications-work-with-nfs-smb-fuse-proc-or-sys) could be an issue? an option was to enable a flag/config to set the "mode" of watcher, implement notify but also the active loop? WDYT?

I don't think this will be an issue with usual local filesystems, but it would be interesting to be able to use the active loop as a escape hatch. This could be added later too.